### PR TITLE
Save property restrictions on nodes/edges

### PR DESCRIPTION
**Pull request by l-hoang**
In Cypher, users can limit matches on matched nodes or edges by
specifying a property as well as what the property should be. For
example, the following node would only match if id was 123:

match (a {id:123})

This commit adds support for saving this information into a map for
later use in query graph IR construction. This will be done added in
a separate commit. Tests have been added to make sure parsed properties
are sane.

### CODE REVIEW

File: libquery/src/CypherCompiler.cpp [hunk 1]
```diff
@@ -172,6 +172,79 @@ unsigned CypherCompiler::GetEdgeID(std::string str) {
...
+      property_value.emplace<0>(cypher_ast_string_get_value(ast_value));
+    } else if (value_type == CYPHER_AST_INTEGER) {
+      property_value.emplace<1>(
+          std::stoi(cypher_ast_integer_get_valuestr(ast_value)));
+    } else if (value_type == CYPHER_AST_FLOAT) {
+      property_value.emplace<2>(
+          std::stof(cypher_ast_float_get_valuestr(ast_value)));
+    } else if (value_type == CYPHER_AST_PARAMETER) {
+      property_value.emplace<3>(cypher_ast_parameter_get_name(ast_value));
+    } else {
```
Comment (roshandathathri) [inline libquery/src/CypherCompiler.cpp:205]: Why not a switch statement?

File: libquery/src/CypherCompiler.cpp [hunk 2]
```diff
@@ -172,6 +172,79 @@ unsigned CypherCompiler::GetEdgeID(std::string str) {
...
+    if (property_value.index() == 0) {
+      GALOIS_LOG_DEBUG("Property: Name {} value {}", property_name,
+                       std::get<0>(property_value));
+    } else if (property_value.index() == 1) {
+      GALOIS_LOG_DEBUG("Property: Name {} value {}", property_name,
+                       std::get<1>(property_value));
+    } else if (property_value.index() == 2) {
+      GALOIS_LOG_DEBUG("Property: Name {} value {}", property_name,
+                       std::get<2>(property_value));
+    } else if (property_value.index() == 3) {
```
Comment (roshandathathri) [inline libquery/src/CypherCompiler.cpp:224]: Ditto

File: libquery/src/CypherCompiler.cpp (hunk 1 above)
Comment (l-hoang) [reply libquery/src/CypherCompiler.cpp:205]: Don't think you can switch unless it's a POD: the type here may not be a POD.

File: libquery/src/CypherCompiler.cpp (hunk 2 above)
Comment (l-hoang) [reply libquery/src/CypherCompiler.cpp:224]: This though a switch statement definitely makes sense; will do.

File: libquery/src/CypherCompiler.cpp (hunk 1 above)
Comment (roshandathathri) [reply libquery/src/CypherCompiler.cpp:205]: Good point. Confirm that the type is not POD.

Comment (l-hoang) [reply libquery/src/CypherCompiler.cpp:205]: The type is indeed a POD, but the type aliases (CYPHER_AST_PARAM and co) are not constants (they're externs from the cypher parser library), so the switch statement doesn't work.

Comment (roshandathathri) [reply libquery/src/CypherCompiler.cpp:205]: Thanks for checking

**Pull request merged**